# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/WebrtcVideoCodecFactory.h
# Commit: 7d270e96dfbf
# Full Hash: 7d270e96dfbfcc5992f9a1704e7d2229d1a645f3
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 4
# Description:
#   Bug 1654112 - Adapt VideoConduit to the new libwebrtc version. r=ng,bwc
#   
#   This hooks up VideoReceiveStream::Config with a decoder factory and
#   VideoSendStream::Config with both a bitrate allocator factory and an encoder
#   factory.
# ==============================================================================

diff -r 2f1e8f5b2fc2 -r 7d270e96dfbf dom/media/webrtc/libwebrtcglue/WebrtcVideoCodecFactory.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/media/webrtc/libwebrtcglue/WebrtcVideoCodecFactory.h	Mon Feb 08 18:15:07 2021 +0100
@@ -0,0 +1,56 @@
+/* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*-*/
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef DOM_MEDIA_WEBRTC_LIBWEBRTCGLUE_WEBRTCVIDEOCODECFACTORY_H_
+#define DOM_MEDIA_WEBRTC_LIBWEBRTCGLUE_WEBRTCVIDEOCODECFACTORY_H_
+
+#include "api/video_codecs/video_decoder_factory.h"
+#include "api/video_codecs/video_encoder_factory.h"
+#include "MediaEventSource.h"
+
+namespace mozilla {
+class WebrtcVideoDecoderFactory : public webrtc::VideoDecoderFactory {
+ public:
+  std::vector<webrtc::SdpVideoFormat> GetSupportedFormats() const override {
+    MOZ_CRASH("Unexpected call");
+    return std::vector<webrtc::SdpVideoFormat>();
+  }
+
+  std::unique_ptr<webrtc::VideoDecoder> CreateVideoDecoder(
+      const webrtc::SdpVideoFormat& aFormat) override;
+};
+
+class WebrtcVideoEncoderFactory : public webrtc::VideoEncoderFactory {
+  class InternalFactory : public webrtc::VideoEncoderFactory {
+   public:
+    std::vector<webrtc::SdpVideoFormat> GetSupportedFormats() const override {
+      MOZ_CRASH("Unexpected call");
+      return std::vector<webrtc::SdpVideoFormat>();
+    }
+
+    std::unique_ptr<webrtc::VideoEncoder> CreateVideoEncoder(
+        const webrtc::SdpVideoFormat& aFormat) override;
+
+    bool Supports(const webrtc::SdpVideoFormat& aFormat);
+  };
+
+ public:
+  WebrtcVideoEncoderFactory()
+      : mInternalFactory(MakeUnique<InternalFactory>()) {}
+
+  std::vector<webrtc::SdpVideoFormat> GetSupportedFormats() const override {
+    MOZ_CRASH("Unexpected call");
+    return std::vector<webrtc::SdpVideoFormat>();
+  }
+
+  std::unique_ptr<webrtc::VideoEncoder> CreateVideoEncoder(
+      const webrtc::SdpVideoFormat& aFormat) override;
+
+ private:
+  const UniquePtr<InternalFactory> mInternalFactory;
+};
+}  // namespace mozilla
+
+#endif